x-pack/legacy/plugins/epm/public/screens/add_data_source/add_data_source_form.tsx

```diff
@@ -28,60 +29,78 @@ const StyledSteps = styled.div`
   }
 `;
 interface AddDataSourceStepsProps {
-  pkgName: string;
-  pkgTitle: string;
-  pkgVersion: string;
-  datasets: Dataset[];
+  package: PackageInfo;
+}
+interface PolicyOption {
+  label: string;
+  value: string;
 }
 export interface FormState {
   datasourceName: string;
   datasets: EuiCheckboxGroupIdToSelectedMap;
+  policies: Array<EuiComboBoxOptionProps<string>>;
 }
+
 const FormNav = styled.div`
   text-align: right;
 `;
 
-export const AddDataSourceForm = (props: AddDataSourceStepsProps) => {
+export const AddDataSourceForm = ({ package: pkg }: AddDataSourceStepsProps) => {
+  const defaultPolicyOption: PolicyOption = { label: 'Default policy', value: 'default' };
+  const [policyOptions, setPolicyOptions] = useState<PolicyOption[]>([defaultPolicyOption]);
+  useEffect(() => {
+    getPolicies()
+      .then(response => response.list)
+      .then(policies => policies.map(policy => ({ label: policy.name, value: policy.id })))
+      .then(setPolicyOptions);
+  }, []);
+
   const [addDataSourceSuccess, setAddDataSourceSuccess] = useState<boolean>(false);
-  const [formState, setFormState] = useState<FormState>({ datasourceName: '', datasets: {} });
+  const [datasourceName, setDatasourceName] = useState<FormState['datasourceName']>('');
```

Comment:
Converted `formState` into an object which had members which used `useState`, instead of being one big `useState` value. That way, each member only had to know how to update their section instead of the whole state object. We still have a `FormState` interface to share.